cmd/frontend/graphqlbackend/code_hosts_test.go

```diff
@@ -0,0 +1,223 @@
+package graphqlbackend
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	mockassert "github.com/derision-test/go-mockgen/testutil/assert"
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbmocks"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc"
+	"github.com/sourcegraph/sourcegraph/internal/types"
+	"github.com/sourcegraph/sourcegraph/lib/pointers"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSchemaResolver_CodeHosts(t *testing.T) {
+	t.Parallel()
+
+	codeHosts := []*types.CodeHost{
+		{ID: 1, URL: "github.com", Kind: extsvc.KindGitHub, APIRateLimitQuota: pointers.Ptr(int32(1)), APIRateLimitIntervalSeconds: pointers.Ptr(int32(1)), GitRateLimitIntervalSeconds: pointers.Ptr(int32(1)), GitRateLimitQuota: pointers.Ptr(int32(1))},
+		{ID: 2, URL: "gitlab.com", Kind: extsvc.KindGitLab, APIRateLimitQuota: pointers.Ptr(int32(2)), APIRateLimitIntervalSeconds: pointers.Ptr(int32(2)), GitRateLimitIntervalSeconds: pointers.Ptr(int32(2)), GitRateLimitQuota: pointers.Ptr(int32(2))},
+		{ID: 3, URL: "bitbucket-cloud.com", Kind: extsvc.KindBitbucketCloud},
+		{ID: 4, URL: "bitbucket.com", Kind: extsvc.KindBitbucketServer, APIRateLimitQuota: pointers.Ptr(int32(4)), APIRateLimitIntervalSeconds: pointers.Ptr(int32(4)), GitRateLimitIntervalSeconds: pointers.Ptr(int32(4)), GitRateLimitQuota: pointers.Ptr(int32(4))},
+	}
+
+	tests := []struct {
+		name  string
+		first int
+		after int32
+	}{
+		{
+			name:  "only first",
+			first: 1,
+			after: 0,
+		},
+		{
+			name:  "first 1 and after 1",
+			first: 1,
+			after: 1,
+		},
+		{
+			name:  "first 1 and after 2",
+			first: 1,
+			after: 2,
+		},
+		{
+			name:  "first 1 and after 3",
+			first: 1,
+			after: 3,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
Nitpick but you could probably get rid of the `name` because it's redundant with the description.

```suggestion
		t.Run(fmt.Sprintf("first=%d after=%d", tc.first, tc.after), func(t *testing.T) {
```